src/Runner.Common/WebsocketFeedServer.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.DistributedTask.WebApi;
+using GitHub.Runner.Sdk;
+using GitHub.Services.Common;
+using GitHub.Services.WebApi.Utilities.Internal;
+
+namespace GitHub.Runner.Common
+{
+    [ServiceLocator(Default = typeof(WebsocketFeedServer))]
+    public interface IWebsocketFeedServer : IRunnerService, IAsyncDisposable
```

Comment:
is this just copy from what's deleted from `IJobServer.cs`?

Reply:
Mostly yes with a slight twist - the fallback with http long poll only works with Actions Service, so I had to refactor that portion a little bit.

Reply:
do we know whether we still need the fallback for the console log?

Reply:
it's a little bit weird that we have `IWebsocketFeedServer`, `IJobServer`, and `IResultServer`, but `IWebsocketFeedServer` somehow depends on `IJobServer`. 😕

Reply:
is it possible to leave the current code in the `IJobServer`, and create a copy of it without the fallback logic control which one to initialize/use inside the `JobServerQueue` base on the new parameter you added to `JobServerQueue.Start()`?
This would be safer for deployment.

Reply:
It should not be that bad, in jobServerQueue, it's basically if/else between JobServer vs. ResultServer.